### PR TITLE
Document DSL macros

### PR DESCRIPTION
**Pull request by james-j-obrien**

### CODE REVIEW

File: flecs_ecs_derive/src/lib.rs [hunk 1]
```diff
@@ -868,7 +868,6 @@ impl Parse for Access {
     }
 }
 
-#[derive(Debug)]
```
Comment (andrewgazelka) [inline flecs_ecs_derive/src/lib.rs:871]: why was this removed? just curious

Comment (james-j-obrien) [reply flecs_ecs_derive/src/lib.rs:871]: Some `syn` types like `LitStr` stopped implementing debug so the derive stopped working. These are all types internal to the macro so there's no user impact.
